src/mccreight.py

```diff
@@ -0,0 +1,142 @@
+class SuffixNode(object):
+
+    def __init__(self, text):
+        self.children = {}
+        self.parent = None
+        self.suffix_link = None
+        self.i_from = 0
+        self.i_to = 0
+        self.text = text
```

Comment:
If this is the string, then make sure that you do not slice anywhere.

It might be more appropriate just to have a link to the string from the suffix tree and get it from there, because then you know that the nodes do not keep separate slices, but there is nothing wrong with having it here. (I do as well in my Python implementation, although in a different way, just because it is more convenient).

Reply:
Okay, thanks for the comments.

Reply:
There is absolutely nothing wrong with what you are doing here. I always get suspicious when I see strings in nodes because I have seen people slice out substrings for edges too many times, but the way you use the string is fine.